http-core/src/main/scala/org/apache/pekko/http/impl/model/parser/SimpleHeaders.scala

```diff
@@ -111,14 +112,14 @@ private[parser] trait SimpleHeaders { this: Parser with CommonRules with CommonA
 
   // http://tools.ietf.org/html/rfc7233#section-4.2
   def `content-range` = rule {
-    (`byte-content-range` | `other-content-range`) ~ EOI ~> (`Content-Range`(_, _))
+    `byte-content-range` ~ EOI ~> (`Content-Range`(_, _)) | `other-content-range` ~ EOI ~> (`Content-Range`(_, _))
```

Comment:
This change is somewhat suspicious, let me look into it

Reply:
this change was part of 0a979b87f0 and indeed back then (on scala 3.1.3) was required to compile. The old syntax seems to work again on scala 3.2.2 and 3.3.0, though, so this change could be reverted.